pyrsistent/_checked_types.py

```diff
@@ -45,18 +47,50 @@ def __str__(self):
             missing_fields=', '.join(self.missing_fields))
 
 
+_preserved_iterables = (
+        Enum,
+)
+"""For types that are themselves iterable, we would mistakenly take their
+elements for the type annotation. This set defines types to 'preserve' by not
+iterating over them."""
+
+
+def maybe_type_to_list(t):
+    """Try to coerce a user-supplied type directive into a list of types."""
+    # TODO FIXME: Move type validation inside this function to consolidate it.
+    # Preserve type annotations that are thmeselves iterable.
+    if isinstance(t, type) and issubclass(t, _preserved_iterables):
+        return [t]
+    # Preserve type annotations given as strings.
+    elif isinstance(t, string_types):
+        return [t]
+    else:
+        if not isinstance(t, Iterable):
```

Comment:
dunno why this clause is here, but I preserved it.

Reply:
You mean the test above? If `t` would be `int` for example calling `list(t)` would raise a `TypeError` but perhaps I don't understand the comment fully.

Reply:
yeah, you're right -- I just missed that.